core/dbt/events/types.py

```diff
@@ -2295,6 +2295,15 @@ def message(self) -> str:
         return f"Set downloads directory='{self.path}'"
 
 
+@dataclass
+class ProjectInvalidName(InfoLevel):
```

Comment:
If this event is only going to be fired in cases like `dbt init invalid-project-name -s`, this might be better suited as an `ErrorLevel` event.